### PR TITLE
Fix changelog generation

### PR DESCRIPTION
**Pull request by rcrozean**
*Issue #, if available:*

*Description of changes:*
Added logic to https://github.com/aws/eks-distro/pull/1907 for selecting prs that match criteria for changelog generation when existing documentation tags don't exist for a particular version. This is particularly useful when calling `make release-docs-limited`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by rcrozean** [comment]
Screenshot 2023-04-04 at 3 58 01 PM

Generated Using the `OVERRIDE_NUMBER=4 make release-docs-limited` command

File: cmd/release/utils/values/searchprs.go [hunk 1]
```diff
@@ -12,34 +12,53 @@ const (
...
+		opts = &github.SearchOptions{Sort: "updated", Order: "asc"}
+		prs, _, err := githubClient.Search.Issues(ctx, "is:pr is:merged 'Bumped Production release number' in:title label:release label:"+releaseVersion, opts)
+		if err != nil {
+			return "", fmt.Errorf("get PRs from %v: %w", githubClient, err)
+		}
+		if overrideNumber == 1 {
+			lastDocRelease = prs.Issues[overrideNumber-1].ClosedAt.Format("2006-01-02T15:04:05+00:00")
+		} else {
+			lastDocRelease = prs.Issues[overrideNumber-1].ClosedAt.Format("2006-01-02T15:04:05+00:00")
+			prevDocRelease = prs.Issues[overrideNumber-2].ClosedAt.Format("2006-01-02T15:04:05+00:00")
```
Comment (kschumy) [inline cmd/release/utils/values/searchprs.go:43]: can we make `"2006-01-02T15:04:05+00:00"` a const?

File: cmd/release/utils/values/searchprs.go [hunk 2]
```diff
@@ -12,34 +12,53 @@ const (
...
+		}
```
Comment (kschumy) [inline cmd/release/utils/values/searchprs.go:44]: Maybe this? Though double check that this is correct. 
```suggestion
		lastDocRelease = prs.Issues[overrideNumber-1].ClosedAt.Format("2006-01-02T15:04:05+00:00")
		if overrideNumber != 1 {
			prevDocRelease = prs.Issues[overrideNumber-2].ClosedAt.Format("2006-01-02T15:04:05+00:00")
		}
```

File: cmd/release/utils/values/searchprs.go [hunk 3]
```diff
@@ -12,34 +12,53 @@ const (
...
-	//Select the most recent pr from the above query and format the date expected for the go-github client
-	lastDocRelease := prs.Issues[0].ClosedAt.Format("2006-01-02T15:04:05+00:00")
+	lastDocRelease := "2006-01-02T15:04:05+00:00"
+	prevDocRelease := "2006-01-02T15:04:05+00:00"
+	if len(prs.Issues) > 0 {
+		//Select the most recent pr from the above query and format the date expected for the go-github client
+		lastDocRelease = prs.Issues[0].ClosedAt.Format("2006-01-02T15:04:05+00:00")
+	} else {
+		//With no document releases we need to be a little bit clever to generate unannounced changelogs.
+		//This finds the
```
Comment (kschumy) [inline cmd/release/utils/values/searchprs.go:33]: Don't leave us hanging on an explanation of the clever solution!

File: cmd/release/utils/values/searchprs.go [hunk 4]
```diff
@@ -12,34 +12,53 @@ const (
...
-func GetChangelogPRs(releaseVersion string) (string, error) {
+func GetChangelogPRs(releaseVersion string, overrideNumber int) (string, error) {
 	githubClient := github.NewClient(nil)
 
 	ctx := context.Background()
-	opts := &github.SearchOptions{}
+	opts := &github.SearchOptions{Sort: "updated"}
 	//Get the date of the last document release for the release version
-	prs, _, err := githubClient.Search.Issues(ctx, "is:pr is:merged label:release label:documentation repo:aws/eks-distro label:" + releaseVersion, opts)
+	prs, _, err := githubClient.Search.Issues(ctx, "is:pr is:merged label:release label:documentation repo:aws/eks-distro label:"+releaseVersion, opts)
```
Comment (kschumy) [inline cmd/release/utils/values/searchprs.go:21]: I still don't get how this works with `label:documentation` but it does. Is it an OR search?

Comment (kschumy) [reply cmd/release/utils/values/searchprs.go:21]: oh wait. Never mind. It's dealt with below

File: cmd/release/utils/values/searchprs.go [hunk 5]
```diff
@@ -12,34 +12,53 @@ const (
...
-		fmt.Sprintf("%v merged:>%v label:base-img-pkg-update label:%v", baseQuery, lastDocRelease, releaseVersion), opts)
+		fmt.Sprintf("%v merged:%v..%v label:base-img-pkg-update label:%v", baseQuery, prevDocRelease, lastDocRelease, releaseVersion), opts)
 	if err != nil {
 		return "", fmt.Errorf("getting base image prs: %w", err)
 	}
 
-	versPRs, _, err := githubClient.Search.Issues(ctx, fmt.Sprintf("%v merged:>%v label:project label:%v",
-		baseQuery, lastDocRelease, releaseVersion), opts)
+	versPRs, _, err := githubClient.Search.Issues(ctx, fmt.Sprintf("%v merged:%v..%v label:project label:%v",
+		baseQuery, prevDocRelease, lastDocRelease, releaseVersion), opts)
```
Comment (kschumy) [inline cmd/release/utils/values/searchprs.go:61]: Nit: can we not use `Doc` the name `prevDocRelease`? There is a 100% chance I'm going to forget how this works and erroneously think it's going off the docs PR

**Pull request merged**